pynamodb/attributes.py

```diff
@@ -288,6 +307,15 @@ def _dynamo_to_python_attr(cls, dynamo_key: str) -> str:
         """
         return cls._dynamo_to_python_attrs.get(dynamo_key, dynamo_key)  # type: ignore
 
+    @classmethod
+    def _get_discriminator_attribute(cls) -> Optional['DiscriminatorAttribute']:
+        return cls.get_attributes()[cls._discriminator] if cls._discriminator else None  # type: ignore
+
+    def _set_discriminator(self) -> None:
+        discriminator_attr = self._get_discriminator_attribute()
+        if discriminator_attr and discriminator_attr.get_discriminator(self.__class__) is not None:
+            self.attribute_values[self._discriminator] = self.__class__  # type: ignore
```

Comment:
`# type: ignore` still needed?

Reply:
same here -- `# type: ignore` is needed since mypy can't find `self._discriminator` created by the metaclass